### PR TITLE
[IMP] mail: introduce ComposerView/saveStateInStore 

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2579306

### CODE REVIEW

**Comment by oomsveta** [comment]
conflicting files

**Comment by alexkuhn** [comment]
Rebased

**Comment by oomsveta** [comment]
conflicting files again 😢

**Review by Nemael — APPROVED** [review]

File: addons/mail/static/src/models/composer_view.js [hunk 1]
```diff
@@ -227,9 +227,7 @@ registerModel({
          * @param {MouseEvent} ev
          */
         onClickEmoji(ev) {
-            if (this.textInputComponent) {
-                this.textInputComponent.saveStateInStore();
```
Comment (seb-odoo) [inline addons/mail/static/src/models/composer_view.js:231]: Could remove this field/useComponentToModel, as it was its only purpose? Unless other PR will need it again?

Comment (alexkuhn) [reply addons/mail/static/src/models/composer_view.js:231]: Pretty sure it's not useful anymore, so I can just remove it in this PR too. Just made the changes.

**Comment by alexkuhn** [comment]
Moved to grouped PR: https://github.com/odoo/odoo/pull/88878

**Pull request closed without merging**